modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/AbstractKeyValueStorage.java

```diff
@@ -224,34 +214,24 @@ public long getCompactionRevision() {
     }
 
     @Override
-    public void startCompaction(long compactionRevision) {
+    public void updateCompactionRevision(long compactionRevision, KeyValueUpdateContext context) {
         assert compactionRevision >= 0 : compactionRevision;
 
         rwLock.writeLock().lock();
 
         try {
             assertCompactionRevisionLessThanCurrent(compactionRevision, rev);
 
+            saveCompactionRevision(compactionRevision, context);
+
             if (isInRecoveryState()) {
-                this.compactionRevision = compactionRevision;
+                setCompactionRevision(compactionRevision);
```

Comment:
I think this step can be done when we start compaction after recovery. So the real `if (isInRecoveryState()) {` here branch should be a no-op.

Reply:
We discussed it in person, let's leave it like that.